packages/@n8n/task-runner/src/js-task-runner/__tests__/js-task-runner.test.ts

```diff
@@ -1342,4 +1342,45 @@ describe('JsTaskRunner', () => {
 			task.cleanup();
 		});
 	});
+
+	describe('prototype pollution prevention', () => {
+		const checkPrototypeIntact = () => {
+			const obj: Record<string, unknown> = {};
+			expect(obj.toString).toBe(Object.prototype.toString);
+			expect(obj.hasOwnProperty).toBe(Object.prototype.hasOwnProperty);
```

Comment:
If the Object prototype has been changed, then the object created here will have the new prototype. Meaning these will always evaluate true. I don't see these to functions being modified, so maybe they could be removed?

Reply:
[3236282](https://github.com/n8n-io/n8n/pull/12588/commits/3236282f8c1edf1639db62560125bf24e6af4883)